src/main/java/org/myrobotlab/service/Servo2.java

```diff
@@ -0,0 +1,1022 @@
+/**
+ *                    
+ * @author GroG (at) myrobotlab.org
+ *  
+ * This file is part of MyRobotLab (http://myrobotlab.org).
+ *
+ * MyRobotLab is free software: you can redistribute it and/or modify
+ * it under the terms of the Apache License 2.0 as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version (subject to the "Classpath" exception
+ * as provided in the LICENSE.txt file that accompanied this code).
+ *
+ * MyRobotLab is distributed in the hope that it will be useful or fun,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * Apache License 2.0 for more details.
+ *
+ * All libraries in thirdParty bundle are subject to their own license
+ * requirements - please refer to http://myrobotlab.org/libraries for 
+ * details.
+ * 
+ * Enjoy !
+ * 
+ * */
+
+package org.myrobotlab.service;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.Timer;
+
+import org.myrobotlab.framework.Service;
+import org.myrobotlab.framework.ServiceType;
+import org.myrobotlab.framework.interfaces.Attachable;
+import org.myrobotlab.framework.interfaces.NameProvider;
+import org.myrobotlab.framework.interfaces.ServiceInterface;
+import org.myrobotlab.logging.Level;
+import org.myrobotlab.logging.LoggerFactory;
+import org.myrobotlab.logging.LoggingFactory;
+import org.myrobotlab.math.Mapper;
+import org.myrobotlab.service.interfaces.ServoControl;
+import org.myrobotlab.service.interfaces.ServoController;
+import org.slf4j.Logger;
+
+/**
+ * @author GroG
+ * 
+ *         Servos have both input and output. Input is usually of the range of
+ *         integers between 0.0 - 180.0, and output can relay those values
+ *         directly to the servo's firmware (Arduino ServoLib, I2C controller,
+ *         etc)
+ * 
+ *         However there can be the occasion that the input comes from a system
+ *         which does not have the same range. Such that input can vary from 0.0
+ *         to 1.0. For example, OpenCV coordinates are often returned in this
+ *         range. When a mapping is needed Servo.map can be used. For this
+ *         mapping Servo.map(0.0, 1.0, 0, 180) might be desired. Reversing input
+ *         would be done with Servo.map(180, 0, 0, 180)
+ * 
+ *         outputY - is the values sent to the firmware, and should not
+ *         necessarily be confused with the inputX which is the input values
+ *         sent to the servo
+ *         
+ *         FIXME - inherit from AbstractMotor ..
+ * 
+ */
+
+public abstract class Servo2 extends Service implements ServoControl {

```

Comment:
If this is abstract, we should call this class AbstractServo ...

Reply:
its not ... I have yet to make the abstract - this is the beginning of an implementation
I'll get Servo worky again as it did ... but I'm not happy with Servo's implementation - So I'm making a Servo2 - they will share the same ServoControl interface, and after I'm happy with that, I'll push common code into an Abstract

Reply:
but.. the class is defined as "public abstract class Servo2..."   if the class isn't supposed to be abstract you should remove the "abstract" in the class definition.

Reply:
Ha !  I think I spent 5 minutes trying to figure out when I added more functions to the interface it inherits why it didn't tell me it had to implement them ... Ahahaha - I think I was trying to do too many things at once..   I do want to create an abstract at some point but not yet ..
Thanks for pointing it out.